DataFormats/TrackerRecHit2D/src/BaseTrackerRecHit.cc

```diff
@@ -6,6 +6,7 @@
 
 
 namespace {
+#ifdef EDM_LM_DEBUG
```

Comment:
Should this be `EDM_ML_DEBUG`?

Reply:
It is the same as the CPP macro name used later in the file:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/TrackerRecHit2D/src/BaseTrackerRecHit.cc#L23

Reply:
I think that this and DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h should have a change to EDM_ML_DEBUG. This appears to be a typo.

About wrapping this solely into just ```EDM_ML_DEBUG```, I note the commented out calls to ```throwExceptionUninitialized``` which are perhaps also needed for other non-```EDM_ML_DEBUG``` debugging.
Similar to TrackerSingleRecHit.cc, I'd like to suggest a flag (e.g. ```DO_THROW_UNITITIALIZED```) and put the currently commented out calls in e.g. ```getKfComponents1D``` behind that flag.